src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -318,21 +352,32 @@ def _find_resources(self, node, requested_cores, requested_gpus,
         # We need to land enough procs on a node such that the cores,
         # lfs and gpus requested per proc is available on the same node
 
-        nprocs = list()
+        num_procs = dict()
+
+        if requested_cores:
+            alloc_cores = min(requested_cores, free_cores)
+            num_procs['cores'] = alloc_cores / core_chunk
+
+        if requested_gpus:
+            alloc_gpus = min(requested_gpus, free_gpus)
+            num_procs['gpus'] = alloc_gpus / gpu_chunk
+
+        if requested_lfs:
+            alloc_lfs = min(requested_lfs, free_lfs)
+            num_procs['lfs'] = alloc_lfs / lfs_chunk
+
+        if requested_mem:
+            alloc_mem = min(requested_mem, free_mem)
+            num_procs['mem'] = alloc_mem / mem_chunk
 
-        if requested_cores: nprocs.append(min(requested_cores, free_cores) / core_chunk)
-        if requested_gpus : nprocs.append(min(requested_gpus,  free_gpus)  / gpu_chunk)
-        if requested_lfs  : nprocs.append(min(requested_lfs,   free_lfs)   / lfs_chunk)
-        if requested_mem  : nprocs.append(min(requested_mem,   free_mem)   / mem_chunk)
 
-        # Find min number of procs determined across lfs, cores, gpus
-        nprocs = min(nprocs)
+        # Find normalized cores, gpus and lfs
+        if requested_cores: alloc_cores = num_procs['cores'] * core_chunk
+        if requested_gpus : alloc_gpus  = num_procs['gpus']  * gpu_chunk
+        if requested_lfs  : lfs         = num_procs['lfs']   * lfs_chunk
+        if requested_mem  : mem         = num_procs['mem']   * mem_chunk
 
-        # Find normalized lfs, cores and gpus
-        if requested_cores: alloc_cores = nprocs * core_chunk
-        if requested_gpus : alloc_gpus  = nprocs * gpu_chunk
-        if requested_lfs  : lfs         = nprocs * lfs_chunk
-        if requested_mem  : mem         = nprocs * mem_chunk
+        self._log.debug('alc : %s %s %s %s', alloc_cores, alloc_gpus, lfs, mem)
```

Comment:
This is a useful message, but there is no way to understand for which unit it is. Does it make sense to add the unit id as well?

Reply:
The `find_resources` method does not know the uid - we'll have to derive that from the surrounding debug messages from the higherlevel scheduler calls.  Since that part of the scheduler code is sequential, this should always be possible.